packages/emails/src/components/ManageLink.tsx

```diff
@@ -1,13 +1,19 @@
-import { getCancelLink, getRescheduleLink } from "@calcom/lib/CalEventParser";
+import { getCancelLink, getRescheduleLink, getBookingUrl } from "@calcom/lib/CalEventParser";
 import type { CalendarEvent, Person } from "@calcom/types/Calendar";
 
 export function ManageLink(props: { calEvent: CalendarEvent; attendee: Person }) {
   // Only the original attendee can make changes to the event
   // Guests cannot
   const t = props.attendee.language.translate;
+  const cancelLink = getCancelLink(props.calEvent);
+  const rescheduleLink = getRescheduleLink(props.calEvent);
+  const bookingLink = getBookingUrl(props.calEvent);
   if (
-    props.attendee.email === props.calEvent.attendees[0]?.email ||
-    props.calEvent.organizer.email === props.attendee.email
+    (props.attendee.email === props.calEvent.attendees[0]?.email ||
+      props.calEvent.organizer.email === props.attendee.email) &&
+    (Boolean(cancelLink) ||
+      (!props.calEvent.recurringEvent && Boolean(rescheduleLink)) ||
+      Boolean(bookingLink))
```

Comment:
We have `!props.calEvent.recurringEvent && Boolean(rescheduleLink))` and then the same on line 37. We could have boolean "showReschedule" and "showCancel" and be used to check if both are false return null and then return rest of actualy content without nesting.

Also, i wonder - i see that we ask `!props.calEvent.recurringEvent && Boolean(rescheduleLink)) ||
      Boolean(bookingLink)` - if rescheduleLink is not given, but bookingLink is, then we still would not show reschedule because on line 37 it does not include `||
      Boolean(bookingLink)`

Reply:
And also abstract `props.calEvent.organizer.email === props.attendee.email` into `const isOrganizer` and same for `const isBooker` - i assume attendees[0] represents person who booked.

Reply:
true, I will rewrite this in a cleaner way